astropy/cosmology/io/yaml.py

```diff
@@ -0,0 +1,174 @@
+
+__all__ = []  # nothing is publicly scoped
+
+
+##############################################################################
+# Serializer Functions
+# these do Cosmology <-> YAML through a modified dictionary representation of
+# the Cosmology object. The Unified-I/O functions are just wrappers to the YAML
+# that calls these functions.
+
+
+def yaml_representer(tag):
+    """:mod:`yaml` representation of |Cosmology| object.
+
+    Parameters
+    ----------
+    tag : str
+        The class tag, e.g. '!astropy.cosmology.LambdaCDM'
+
+    Returns
+    -------
+    representer : callable[[`~astropy.io.misc.yaml.AstropyDumper`, |Cosmology|], str]
+        Function to construct :mod:`yaml` representation of |Cosmology| object.
+    """
+    def representer(dumper, obj):
+        """Cosmology yaml representer function.
+
+        Parameters
+        ----------
+        dumper : `~astropy.io.misc.yaml.AstropyDumper`
+        obj : `~astropy.cosmology.Cosmology`
+
+        Returns
+        -------
+        str
+            :mod:`yaml` representation of |Cosmology| object.
+        """
+        # convert to mapping
+        map = obj.to_format("mapping")
+        # remove the cosmology class info. It's already recorded in `tag`
+        map.pop("cosmology")
+        # make the metadata serializable in a lossless way
+        map["meta"] = tuple(map["meta"].items())
+
+        return dumper.represent_mapping(tag, map)
+
+    return representer
+
+
+def yaml_constructor(cls):
+    """Cosmology| object from :mod:`yaml` representation.
+
+    Parameters
+    ----------
+    cls : type
+        The class type, e.g. `~astropy.cosmology.LambdaCDM`.
+
+    Returns
+    -------
+    constructor : callable
+        Function to construct |Cosmology| object from :mod:`yaml` representation.
+    """
+    def constructor(loader, node):
+        """Cosmology yaml constructor function.
+
+        Parameters
+        ----------
+        loader : `~astropy.io.misc.yaml.AstropyLoader`
+        node : `yaml.nodes.MappingNode`
+            yaml representation of |Cosmology| object.
+
+        Returns
+        -------
+        `~astropy.cosmology.Cosmology` subclass instance
+        """
+        # create mapping from YAML node
+        map = loader.construct_mapping(node)
+        # restore metadata to dict
+        map["meta"] = dict(map["meta"])
+        # get cosmology class qualified name from node
+        cosmology = str(node.tag).split(".")[-1]
+        # create Cosmology from mapping
+        return cls.from_format(map, format="mapping", cosmology=cosmology)
+
+    return constructor
+
+
+def register_cosmology_yaml(cosmo_cls):
+    """Register :mod:`yaml` for Cosmology class.
+
+    Parameters
+    ----------
+    cosmo_cls : `~astropy.cosmology.Cosmology` class
+    """
+    tag = f"!{cosmo_cls.__module__}.{cosmo_cls.__qualname__}"
+
+    AstropyDumper.add_multi_representer(cosmo_cls, yaml_representer(tag))
```

Comment:
Does this need to be a `multi_representer` now if we are just registering all known cosmology classes?

Reply:
It doesn't! just checked.

Reply:
Done!